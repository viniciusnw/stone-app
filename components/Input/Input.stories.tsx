import React from "react";
import { Input, StorybookWrapper } from "@Components";

export const Default = {};

export default {
  title: "Input",
  component: (props: any) => (
    <StorybookWrapper>
      <Input value="" />
    </StorybookWrapper>
  ),
};
