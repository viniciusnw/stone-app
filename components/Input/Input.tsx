import React from 'react';
import styled from 'styled-components/native';
import {TextInput} from 'react-native';

const StyledInput = styled(TextInput)`
  width: 100%;
  height: 42px;
  border: 1px solid ${({theme}) => theme.color.border};
  padding: 10px 15px;
`;

const Input = (props: any) => {
  return <StyledInput {...props} />;
};

export default Input;
