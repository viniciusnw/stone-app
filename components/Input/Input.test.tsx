import React from "react";
import renderer from "react-test-renderer";
import { darkTheme, lightTheme } from "@Theme";
import { ThemeProvider } from "styled-components/native";
import "jest-styled-components";
import Input from "./Input";

jest.useFakeTimers();

test("Input", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={darkTheme}>
        <Input value="" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
