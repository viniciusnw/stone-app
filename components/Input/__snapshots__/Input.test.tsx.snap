// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input 1`] = `
<TextInput
  style={
    {
      "borderColor": "#ddd",
      "borderStyle": "solid",
      "borderWidth": 1,
      "height": 42,
      "paddingBottom": 10,
      "paddingLeft": 15,
      "paddingRight": 15,
      "paddingTop": 10,
      "width": "100%",
    }
  }
  value=""
/>
`;
